### ISSUE TITLE
Dependency Installation Issues and Version Conflicts

### ISSUE THREAD

**Issue by ethahtz** [princeton-nlp/Edge-Pruning#4]
Hi,

I encountered the following issues during installation (I am using python3.11):
```
ERROR: Could not install packages due to an OSError: [Errno 2] No such file or directory: '/scratch/gpfs/ab4197/p-printer/tracr/tracr'  
```
To address it, I switched to using tracr directly from its original GitHub repo:
```
tracr @ git+https://github.com/deepmind/tracr  
```
This resolved the problem, but it led to the following dependency conflict.

```
ERROR: Ignored the following yanked versions: 2.0.0  
ERROR: Could not find a version that satisfies the requirement torchaudio==2.0.2+cu118 (from versions: 2.0.1, 2.0.2, 2.1.0, ...)  
ERROR: No matching distribution found for torchaudio==2.0.2+cu118  
```
To proceed, I attempted to remove torchaudio from the requirements file, as it seems unnecessary for pruning subnetworks.

After removing torchaudio, I ran into further dependency conflicts with accelerate:

```
ERROR: Cannot install -r requirements.txt (line 116) and accelerate==0.28.0 because these package versions have conflicting dependencies.  

The conflict is caused by:
    The user requested accelerate==0.28.0  
    llm-foundry 0.3.0 depends on accelerate<0.21 and >=0.20  
```

At this point, it feels like there are too many version conflicts to resolve without additional guidance. Could you please recommend the best approach for resolving these conflicts?

Thank you in advance for your help!

**Comment by testzer0** [comment]
You're right that the `requirements.txt` file is too bloated. I will attempt to cut it down. In the meantime, could you try the following? The only library whose version should matter is `transformers==4.40.0`, as we depend on this version for our modified modeling files. Otherwise, any version of accelerate compatible with this should work fine.
